### DISCUSSION TITLE
API implementation for graph image download

### DISCUSSION THREAD

**Discussion by kornix** [akvorado/akvorado#97] [Q&A]
Hi all.

I would like to offer an idea to implement an API to download a graph as an image to display it somewhere else or to integrate feature rich graph display to another site. I mean something like Observium's https://<hostname>/graph.php?<options> way which returns data with "image/png" or "image/svg" content type, so it can be used further simply like "<img src='https://<hostname>/graph.php?<options>'>" somewhere else.

I think feature allowing to expose data from single source to another resources would be very useful (at least, i would use it for sure).

Thank you in advance.

**Comment by vincentbernat** [comment]
Observium is using RRD tool, so images are generated server-side. Here, the images are generated by the client. It could be possible with an headless browser. So, no easy way to implement that yet.

**Reply by kornix** [reply]
Yes, it's understood. I think I'll try to do that conversation with puppeteer or something similar. If the result will be acceptable, I'll share the recipe. Maybe it will come in handy in version 2.x.x :) If no, it still could be used as is (apparently, I prefer calling /api/v0/graph to get desired result :) ).
